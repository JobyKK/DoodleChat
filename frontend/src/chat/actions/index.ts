import { call, put, race, delay, take } from 'redux-saga/effects';

export enum ACTION_TYPE {
    SEND_USER_MESSAGE = 'SEND_USER_MESSAGE',
    RESEIVE_MESSAGES = 'RESEIVE_MESSAGES',
    RESEIVE_MESSAGES_FAILURE = 'RESEIVE_MESSAGES_FAILURE',
    POLL_START = 'POLL_START',
    POLL_STOP = 'POLL_STOP'
}

export const startPollMessages = () => {
    return {
        type: ACTION_TYPE.POLL_START
    }
}

export const stopPollMessages = () => {
    return {
        type: ACTION_TYPE.POLL_STOP
    }
}

export const sendMessage = (message) => (dispatch) => {
    fetch("http://localhost:8080/api/message", {
        method: 'POST',
        mode: 'cors',
        headers: {
            'Accept': 'application/json',
            'Content-Type': 'application/json'
        },
        body: JSON.stringify(message)
    })
    .then(res => {
        dispatch(() => ({
            type: ACTION_TYPE.SEND_USER_MESSAGE
        }))
    })
    .catch(err => console.log(err));
};
