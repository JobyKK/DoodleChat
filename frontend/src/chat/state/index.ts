export interface Owner {
    id: number;
    name?: String;
}

export interface Message {
    text: string;
    date: Date;
    owner: Owner;
}

export default class ChatState {
    messages: Array<Message>;

    public static getMessages = (state) => state.chat.messages;
}