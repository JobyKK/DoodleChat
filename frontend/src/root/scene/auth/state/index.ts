export interface User {
    id: number;
    name: String;
}

export class UserState {
    private user: User;

    public static getUser = (state) => state.auth.user;
}
