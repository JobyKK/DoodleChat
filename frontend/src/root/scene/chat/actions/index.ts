let ws: WebSocket;

export enum ACTION_TYPE {
    SEND_USER_MESSAGE = 'SEND_USER_MESSAGE',
    RESEIVE_MESSAGES = 'RESEIVE_MESSAGES',
    RESEIVE_MESSAGES_FAILURE = 'RESEIVE_MESSAGES_FAILURE',
    WS_CONNECTED = 'WS_CONNECTED'
}

export const subscribeOnMessages = () => (dispatch) => {
    if (ws != null) {
        return {type: ACTION_TYPE.WS_CONNECTED};
    }

    ws = new WebSocket('ws://localhost:8080/chat');

    ws.onmessage = (message) => {
        dispatch({
            type: ACTION_TYPE.RESEIVE_MESSAGES,
            messages: JSON.parse(message.data)
        })
    }
}

export const sendMessage = (message) => (dispatch) => {
    ws.send(JSON.stringify(message));
};
