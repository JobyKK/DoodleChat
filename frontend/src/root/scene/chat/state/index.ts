export interface Owner {
    id?: number;
    name?: String;
}

export interface Message {
    id: number;
    text: string;
    date: Date;
    owner: Owner;
}

export default class ChatState {
    messages: Array<Message>;

    public static getMessages = (state) => state.chat.messages;
}