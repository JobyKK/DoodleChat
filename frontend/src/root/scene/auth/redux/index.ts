import { combineReducers } from 'redux';
import { ACTION_TYPE } from '../actions';
import { UserState } from '../state';

const authReducer = (state = UserState, action: any) => {
    switch (action.type) {
        case ACTION_TYPE.LOGIN_USER:
            return {
                ...state,
                user: action.user
            };
        default:
            return state;
    }
};

export default authReducer;
