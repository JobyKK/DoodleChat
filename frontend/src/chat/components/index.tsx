import * as React from 'react';
import { WithStyles, withStyles } from '@material-ui/core/styles';
import { styles } from './styles/index';
import { User } from '../scene/auth/state';
import { Message } from '../state';
import InputBlock from './common/InputBlock';
import AuthBlock from './../scene/auth/containers';
import MessageList from './messages';

interface ChatProps {
    user: User;
    messages: Array<Message>;
    sendMessage: (message: Message) => void;
    subscribeOnMessages: any;
    getAllMessages: any;
}

export const Chat = (props: ChatProps & WithStyles<any>) => {
    React.useEffect(() => {
        props.subscribeOnMessages();
    }, [props.user])

    let sendMessage = (text: string) => props.sendMessage({
        text,
        date: new Date(),
        owner: { id: props.user.id }
    } as Message)

    return (
        <div className={props.classes.container}>
            <MessageList user={props.user} messages={props.messages}/>
            {
                props.user
                    ? <InputBlock onSubmit={sendMessage} buttonText={'Send'}
                        inputPlaceholder={'Message'} />
                    : <AuthBlock />
            }
        </div>
    )
};

export default withStyles(styles)(Chat);
