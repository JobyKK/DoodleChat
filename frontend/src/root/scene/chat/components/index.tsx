import * as React from 'react';
import { WithStyles, withStyles } from '@material-ui/core/styles';
import { styles } from './styles/index';
import { User } from '../../auth/state';
import { Message } from './../state';
import InputBlock from '../../../components/common/InputBlock';
import MessageList from './messages';

export interface ChatProps {
    user: User;
    messages: Array<Message>;
    sendMessage: (message: Message) => void;
    subscribeOnMessages: any;
    getAllMessages: any;
}

const Chat = (props: ChatProps & WithStyles<any>) => {
    React.useEffect(() => {
        props.subscribeOnMessages();
    }, [props.user])

    let sendMessage = (text: string) => props.sendMessage({
        text,
        date: new Date(),
        owner: { id: props.user.id }
    } as Message)

    return (
        <React.Fragment>
            <MessageList user={props.user} messages={props.messages}/>
            {
                props.user && <InputBlock onSubmit={sendMessage} buttonText={'Send'}
                        inputPlaceholder={'Message'} />
            }
        </React.Fragment>
    )
};

export default withStyles(styles)(Chat);
