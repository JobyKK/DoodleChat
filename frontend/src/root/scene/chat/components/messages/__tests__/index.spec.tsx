import * as React from 'react';
import { MessageList, MessageListProps } from './../index';
import { User } from 'src/chat/scene/auth/state';
import renderer from 'react-test-renderer';
import { Message, Owner } from 'src/chat/state';

let props = {} as MessageListProps;

beforeEach(() => {
  props = {
    user: {id: 1, name: "Name"} as User,
    messages: [],
    classes: {} 
  } as MessageListProps;
});

describe('MessageList component', () => {
  it('draw my message', () => {
    props.messages = [
      {
        id: 1,
        text: "text1",
        date: new Date(),
        owner: props.user
      } as Message
    ]

    const wrapper = renderer.create(<MessageList {...props} />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it('draw a chat message', () => {
    props.messages = [
      {
        id: 1,
        text: "text1",
        date: new Date(),
        owner: {
          id: 321, name: "Name2"
        } as Owner
      } as Message
    ]

    const wrapper = renderer.create(<MessageList {...props} />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
