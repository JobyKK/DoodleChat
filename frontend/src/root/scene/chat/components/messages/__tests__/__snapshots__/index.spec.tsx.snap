// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList component draw chat message 1`] = `
<div>
  <section
    className="MessageBlock-container-7"
  >
    <div
      className="MessageBlock-userNameLabel-10"
    >
      Name2
    </div>
    <div
      className="MessageBlock-textLabel-11"
    >
      text1
    </div>
    <div
      className="MessageBlock-dateLabel-12"
    >
      Apr 10, 2019, 19:53
    </div>
  </section>
</div>
`;

exports[`MessageList component draw my message 1`] = `
<div>
  <section
    className="MyMessageBlock-myMessageContainer-3"
  >
    <section
      className="MessageBlock-container-7 MyMessageBlock-container-1"
    >
      <div
        className="MessageBlock-userNameLabel-10 MyMessageBlock-userNameLabel-4"
      >
        You
      </div>
      <div
        className="MessageBlock-textLabel-11 MyMessageBlock-textLabel-5"
      >
        text1
      </div>
      <div
        className="MessageBlock-dateLabel-12 MyMessageBlock-dateLabel-6"
      >
        Apr 10, 2019, 19:53
      </div>
    </section>
  </section>
</div>
`;
